sources/nvidia-migmanager/src/main.rs

```diff
@@ -0,0 +1,685 @@
+
+#[derive(Debug, PartialEq)]
+enum NvidiaGpu {
+    A100_40GB,
+    A100_80GB,
+    H100_80GB,
+    H200_141GB,
+    Other,
+}
+
+#[derive(Hash, Debug, Clone, PartialEq, Eq)]
+enum MigState {
+    Unsupported,
+    Enabled,
+    Disabled,
+    Transition,
+    Unknown,
+}
+
+struct MigGpu {
+    model: NvidiaGpu,
+    state: MigState,
+}
+
+/// Wrapper around process::Command that adds error checking.
+fn command<I, S>(bin_path: &str, args: I) -> Result<String>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let mut command = Command::new(bin_path);
+    command.args(args);
+    let output = command
+        .output()
+        .context(error::ExecutionFailureSnafu { command })?;
+
+    trace!("stdout: {}", String::from_utf8_lossy(&output.stdout));
+    trace!("stderr: {}", String::from_utf8_lossy(&output.stderr));
+
+    ensure!(
+        output.status.success(),
+        error::CommandFailureSnafu { bin_path, output }
+    );
+
+    let output_str = String::from_utf8_lossy(&output.stdout);
+
+    Ok(output_str.to_string())
+}
+
+// Runs the nvidia-smi command to enable/disable MIG in all GPUs
+fn set_mig_mode(mig_enabled: bool) -> Result<()> {
+    info!(
+        "{} MIG.",
+        if mig_enabled { "Enabling" } else { "Disabling" }
+    );
+
+    command(NVIDIA_SMI_PATH, ["-mig", &(mig_enabled as u8).to_string()])?;
+    Ok(())
+}
+
+// Runs the nvidia-smi command to apply the correct MIG profile in all the GPUs
+fn set_mig_profile(gpu_indices: Vec<usize>, profile_string: &str) -> Result<()> {
+    info!("Activating MIG profile.");
+
+    let indices = gpu_indices
+        .iter()
+        .map(|n| n.to_string())
+        .collect::<Vec<String>>()
+        .join(",");
+
+    command(
+        NVIDIA_SMI_PATH,
+        ["-i", indices.as_str(), "mig", "-cgi", profile_string, "-C"],
+    )?;
+    Ok(())
+}
+
+// Uses pci-device id to find out the GPU model of the instance
+fn get_gpu_model(pci_device_id: &str) -> Result<NvidiaGpu> {
+    ensure!(
+        pci_device_id.ends_with(NVIDIA_VENDOR_ID),
+        error::GpuModelSnafu
+    );
+
+    if pci_device_id.starts_with("0x20B0") {
+        info!("Found NVIDIA A100-40GB GPU.");
+        Ok(NvidiaGpu::A100_40GB)
+    } else if pci_device_id.starts_with("0x20B2") || pci_device_id.starts_with("0x20B5") {
+        info!("Found NVIDIA A100-80GB GPU.");
+        Ok(NvidiaGpu::A100_80GB)
+    } else if pci_device_id.starts_with("0x2330")
+        || pci_device_id.starts_with("0x2321")
+        || pci_device_id.starts_with("0x2331")
+        || pci_device_id.starts_with("0x2339")
+    {
+        info!("Found NVIDIA H100-80GB GPU.");
+        Ok(NvidiaGpu::H100_80GB)
+    } else if pci_device_id.starts_with("0x2335")
+        || pci_device_id.starts_with("0x233B")
+        || pci_device_id.starts_with("0x2348")
+    {
+        info!("Found NVIDIA H200-141GB GPU.");
+        Ok(NvidiaGpu::H200_141GB)
+    } else {
+        info!("Found NVIDIA Device but couldn't confirm variant.");
```

Comment:
This to me justifies a warning right? 

```suggestion
        warn!("Found NVIDIA Device but couldn't confirm variant.");
```